src/tests/performance/get_blocks.rs

```diff
@@ -0,0 +1,197 @@
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     600 │        100 │          0 │         14 │              2 │          1 │          1 │          2 │          2 │          6 │          99.00 │       7.46 │      7961.44 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     700 │        100 │          0 │         11 │              2 │          1 │          1 │          2 │          3 │          6 │          98.77 │      10.43 │      6629.62 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     800 │        100 │          0 │         11 │              1 │          1 │          1 │          2 │          2 │          5 │          99.02 │       7.62 │     10396.44 │
+    // └─────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴────────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴──────────────┘
+    // *NOTE* run with `cargo test --release  tests::performance::get_blocks -- --nocapture`
+    // Before running test generate dummy devices with different ips using toos/ips.py
+
+    let synth_counts = vec![1, 50, 100, 200, 300, 400, 500, 600, 700, 800];
+    //let synth_counts = vec![1, 50,];
+
+    let mut table = RequestsTable::default();
+
+    for synth_count in synth_counts {
+        let target = TempDir::new().expect("Unable to create TempDir");
+        let mut node = Node::builder()
+            .build(target.path())
+            .expect("unable to build the node");
+        node.start().await;
+
+        let node_addr = node.net_addr().expect("network address not found");
+
+        let mut synth_sockets = Vec::with_capacity(synth_count);
+        let mut ips = IPS.to_vec();
+
+        for _ in 0..synth_count {
+            let socket = TcpSocket::new_v4().unwrap();
+
+            // Make sure we can reuse the address and port
+            socket.set_reuseaddr(true).unwrap();
+            socket.set_reuseport(true).unwrap();
+
+            // If there is address for our thread in the pool we can use it.
+            // Otherwise we'll not set bound_addr and use local IP addr (127.0.0.1).
+            let ip = if let Some(ip_addr) = ips.pop() {
+                SocketAddr::new(
+                    IpAddr::V4(Ipv4Addr::from_str(ip_addr).unwrap()),
+                    CONNECTION_PORT,
+                )
+            } else {
+                "127.0.0.1:0".parse().unwrap()
+            };

```

Comment:
hmhmh, what about something like this (haven't tried to compile it, but should be easily adjusted if compilation breaks)
```rust
   let (ip, port) = if let Some(ip_addr) = ips.pop() {
        (ip_addr, CONNECTION_PORT)
    } else {
        ("127.0.0.1", 0)
    };

   let ip = IpAddr::V4(Ipv4Addr::from_str(ip).unwrap();
   let socket = SocketAddr::new(ip, port)
```

Reply:
could be even more simplified if we could always use the number zero for the port...

Could we? Why do we want to use that odd number:
```rust
const CONNECTION_PORT: u16 = 31337;
```

Reply:
Solved.